test/lib/y2network/proposal_settings_test.rb

```diff
@@ -140,26 +173,62 @@ def stub_features(features)
       subject.enable_wicked!
     end
 
-    it "sets :wicked as the backend" do
-      expect(subject.backend).to eql(:network_manager)
+    it "sets :wicked as the user selected backend" do
+      expect(subject.selected_backend).to be_nil
+      subject.enable_wicked!
+      expect(subject.selected_backend).to eql(:wicked)
     end
   end
 
-  describe "#enable_network_manager!" do
-    it "adds the NetworkManager package to the list of resolvables " do
-      expect(Yast::PackagesProposal).to receive(:AddResolvables)
-        .with("network", :package, ["NetworkManager"])
-      subject.enable_network_manager!
+  describe "#refresh_packages" do
+    let(:backend) { :wicked }
+
+    before do
+      allow(subject).to receive(:current_backend).and_return(backend)
     end
 
-    it "removes the wicked package from the list of resolvables " do
-      expect(Yast::PackagesProposal).to receive(:RemoveResolvables)
-        .with("network", :package, ["wicked"])
-      subject.enable_network_manager!
+    context "when :wicked is the current backend" do
+      it "adds the wicked package to the list of resolvables " do
+        expect(Yast::PackagesProposal).to receive(:AddResolvables)
+          .with("network", :package, ["wicked"])
+        subject.refresh_packages
+      end
+
+      it "removes the NetworkManager package from the list of resolvables " do
+        expect(Yast::PackagesProposal).to receive(:RemoveResolvables)
+          .with("network", :package, ["NetworkManager"])
+        subject.refresh_packages
+      end
     end
 
-    it "sets :network_manager as the backend" do
-      expect(subject.backend).to eql(:network_manager)
+    context "when :network_manager is the current backend" do
+      let(:backend) { :network_manager }
+
+      it "adds the NetworkManager package to the list of resolvables " do
+        expect(Yast::PackagesProposal).to receive(:AddResolvables)
+          .with("network", :package, ["NetworkManager"])
+        subject.refresh_packages
+      end
+
+      it "removes the wicked package from the list of resolvables " do
+        expect(Yast::PackagesProposal).to receive(:RemoveResolvables)
+          .with("network", :package, ["wicked"])
+        subject.refresh_packages
+      end
+    end
+  end
+
+  describe "#enable_network_manager!" do
+    before do
+      subject
+    end
```

Comment:
You can use a single `before do` :wink:

Reply:
These are really leftovers :+1: updated!